src/amo/components/AddonMoreInfo/index.js

```diff
@@ -326,6 +395,9 @@ export const mapStateToProps = (state: AppState, ownProps: Props) => {
     versionInfo,
     hasStatsPermission: hasPermission(state, STATS_VIEW),
     userId: state.users.currentUserID,
+    clientApp: state.api.clientApp,
+    loading: state.categories.loading,
+    categoriesState: state.categories.categories,
```

Comment:
Instead of returning `state.categories.categories`, in this `mapStateToProps` function you could call the new function that I discussed above, `getCategoryNames` (which you will create in the `categories` reducer, and return the list of category names in `relatedCategories`.